advanced-achievements-plugin/src/main/java/com/hm/achievement/command/executable/InspectCommand.java

```diff
@@ -0,0 +1,166 @@
+			return;
+		}
+		int page = getPage(args);
+
+		advancedAchievements.getServer().getScheduler().runTaskAsynchronously(advancedAchievements, () -> {
+			// Cleaning the cache & caching desired pagination
+			cleanUpCache();
+			checkAndCache(achievementName);
+
+			// Send pagination
+			SupplierCommandPagination pagination = cachedPaginations.get(achievementName);
+			pagination.sendPage(page, sender);
+		});
+	}
+
+	private String parseAchievementName(String[] args) {
+		StrBuilder achName = new StrBuilder();
+
+		boolean lastArgumentIsNumber = args.length > 1 && NumberUtils.isDigits(args[args.length - 1]);
+		boolean secondLastArgumentIsNumber = args.length > 2 && NumberUtils.isDigits(args[args.length - 2]);
+		if (lastArgumentIsNumber && secondLastArgumentIsNumber) {
+			List<String> endsRemoved = new ArrayList<>(Arrays.asList(args).subList(1, args.length - 1));
+			achName.appendWithSeparators(endsRemoved, " ");
+		} else {
+			List<String> firstArgRemoved = new ArrayList<>(Arrays.asList(args).subList(1, args.length));
+			achName.appendWithSeparators(firstArgRemoved, " ");
+		}
+
+		return achName.toString();
+	}
+
+	private String getAchievementName(String achievementDisplayName) {
+		// Initialize this map here, in constructor achievementsAndDisplayNames is empty
+		if (achievementDisplayNamesAndDatabaseNames == null) {
+			achievementDisplayNamesAndDatabaseNames = reverseMap(achievementsAndDisplayNames);
+		}
+
+		String achievementName = achievementDisplayNamesAndDatabaseNames.get(achievementDisplayName);
+		if (achievementName == null) {
+			String whiteSpaceRemovedName = achievementDisplayName.replaceAll("\\s", "_");
+			if (achievementDisplayNamesAndDatabaseNames.values().contains(whiteSpaceRemovedName)) {
+				achievementName = whiteSpaceRemovedName;
+			}
+		}
+		return achievementName;
+	}
+
+	private void cleanUpCache() {
+		long time = System.currentTimeMillis();
+		Set<String> toRemove = new HashSet<>();
+		for (Map.Entry<String, Long> entry : lastCached.entrySet()) {
+			if (time - CACHE_EXPIRATION_DELAY > entry.getValue()) {
+				toRemove.add(entry.getKey());
+			}
+		}
+		for (String achievementName : toRemove) {
+			cachedPaginations.remove(achievementName);
+			lastCached.remove(achievementName);
+		}
+
+	}
+
+	private void checkAndCache(String achievementName) {
+		if (System.currentTimeMillis() - CACHE_EXPIRATION_DELAY > lastCached.getOrDefault(achievementName, 0L)) {
+			List<AwardedDBAchievement> recipientList = databaseManager.getAchievementsRecipientList(achievementName);
+			// Use Suppliers to avoid huge work on getting UUID - name relations.
+			List<Supplier<String>> messages = recipientList.stream()
+					.map(achievement -> (Supplier<String>) () -> {
+						UUID uuid = achievement.getAwardedTo();
+						OfflinePlayer player = advancedAchievements.getServer().getOfflinePlayer(uuid);
+						String identifier = player != null ? player.getName() : uuid.toString();
```

Comment:
I believe `OfflinePlayer` will never be null. `player.getName()` can be null though.